packages/core/src/notification.ts

```diff
@@ -0,0 +1,592 @@
+  UNKNOWN: 'UNKNOWN'
+}
+
+Object.freeze(notificationCategory)
+const categories = Object.values(notificationCategory)
+
+type NotificationClassification = NotificationCategory
+/**
+ * @typedef {NotificationCategory} NotificationClassification
+ * @experimental
+ */
+/**
+ * Constants that represents the Classification in the {@link GqlStatusObject}
+ * @type {notificationCategory}
+ * @experimental
+ */
+const notificationClassification = notificationCategory
+
+/**
+ * Class for Cypher notifications
+ * @access public
+ */
+class Notification {
+  code: string
+  title: string
+  description: string
+  severity: string
+  position: NotificationPosition | {}
+  severityLevel: NotificationSeverityLevel
+  category: NotificationCategory
+  rawSeverityLevel: string
+  rawCategory?: string
+
+  /**
+   * Create a Notification instance
+   * @constructor
+   * @param {Object} notification - Object with notification data
+   */
+  constructor (notification: any) {
+    /**
+     * The code
+     * @type {string}
+     * @public
+     */
+    this.code = notification.code
+    /**
+     * The title
+     * @type {string}
+     * @public
+     */
+    this.title = notification.title
+    /**
+     * The description
+     * @type {string}
+     * @public
+     */
+    this.description = notification.description
+    /**
+     * The raw severity
+     *
+     * Use {@link Notification#rawSeverityLevel} for the raw value or {@link Notification#severityLevel} for an enumerated value.
+     *
+     * @type {string}
+     * @public
+     * @deprecated This property will be removed in 6.0.
+     */
+    this.severity = notification.severity
+    /**
+     * The position which the notification had occur.
+     *
+     * @type {NotificationPosition}
+     * @public
+     */
+    this.position = _constructPosition(notification.position)
+
+    /**
+     * The severity level
+     *
+     * @type {NotificationSeverityLevel}
+     * @public
+     * @example
+     * const { summary } = await session.run("RETURN 1")
+     *
+     * for (const notification of summary.notifications) {
+     *     switch(notification.severityLevel) {
+     *         case neo4j.notificationSeverityLevel.INFORMATION: // or simply 'INFORMATION'
+     *             console.info(`${notification.title} - ${notification.description}`)
+     *             break
+     *         case neo4j.notificationSeverityLevel.WARNING: // or simply 'WARNING'
+     *             console.warn(`${notification.title} - ${notification.description}`)
+     *             break
+     *         case neo4j.notificationSeverityLevel.UNKNOWN: // or simply 'UNKNOWN'
+     *         default:
+     *             // the raw info came from the server could be found at notification.rawSeverityLevel
+     *             console.log(`${notification.title} - ${notification.description}`)
+     *             break
+     *     }
+     * }
+     */
+    this.severityLevel = _asEnumerableSeverity(this.severity)
```

Comment:
```suggestion
    this.severityLevel = _asEnumerableSeverity(notification.severity)
```
nit: potential for spooky action at a distance.

Reply:
true, better get the information from the notification instead of this, this way if we change the meaning of this.severity (or delete), severityLevel will not break